sopel/privileges.py

```diff
@@ -1,146 +1,147 @@
-"""Constants for user privileges in channels.
+"""Constants for user privileges in channels."""
+# Copyright 2023 dgw, technobabbl.es
+from __future__ import annotations
 
-Privilege levels
-================
+import enum
 
-Historically, there were two user privileges in channels:
 
-* :data:`OP`: channel operator, or chanop, set and unset by ``+o`` and ``-o``
-* :data:`VOICE`: the privilege to send messages to a channel with the
-  ``+m`` mode, set and unset by ``+v`` and ``-v``
+__all__ = [
+    'AccessLevel',
+]
 
-Since then, other privileges have been adopted by IRC servers and clients:
 
-* :data:`HALFOP`: intermediate level between Voiced and OP, set and unset by
-  ``+h`` and ``-h``
-* :data:`ADMIN`: channel admin, above OP and below OWNER, set and unset by
-  ``+a`` and ``-a``
-* :data:`OWNER`: channel owner, above ADMIN and OP, set and unset by ``+q`` and
-  ``-q``
+class AccessLevel(enum.IntFlag):
+    """Enumeration of available user privilege levels.
 
-.. important::
+    Comparing privileges
+    ====================
 
-    Not all IRC networks support these added privilege modes. If you are
-    writing a plugin for public distribution, ensure your code behaves sensibly
-    if only +v (voice) and +o (op) modes exist.
+    This class represents privileges as powers of two, with higher values
+    assigned to higher-level privileges::
```

Comment:
Nit: I'm not sure how I feel about recapitulating the behavior of `auto()` here, it seems like a detail that could just as easily be omitted.

Reply:
A few lines below you can see the original documentation for the module-level constants, which had this exact language. There's an argument to be made that this whole introduction to the idea of privilege flags can be improved, but I hope you don't pin the wording on me. I just moved around what was already there. 😝

If you really want to take care of it now, I'll see what I can come up with later in the week.

Reply:
Psych, I had a brainstorm tonight after all.